File: BasicInputOutput/Java/PowerOfTwoBitwise.java

import java.util.Scanner;

class PowerOfTwoBitwise{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Enter a value : ");
        int num = sc.nextInt();

        int check = num & (num-1);
        if(check==0){
            System.out.println(num+" is Power of 2 ");
        }else{
            System.out.println(num+" is not Power of 2 ");
        }
    }
}

File: BasicInputOutput/C/PowerOfTwoBitwise.c
#include <stdio.h>

int main() {
    int num;
    printf("Enter a value: ");
    scanf("%d", &num);

    int check = num & (num - 1);
    if (check == 0) {
        printf("%d is a Power of 2\n", num);
    } else {
        printf("%d is not a Power of 2\n", num);
    }

    return 0;
}
